core/src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -216,6 +217,21 @@ object DoublePassOTreeDataAnalyzer
       indexStatus: IndexStatus,
       options: QbeastOptions): (DataFrame, TableChanges) = {
     logTrace(s"Begin: Analyzing the input data with existing revision: ${indexStatus.revision}")
+
+    // Check if the DataFrame is deterministic
+    logDebug(s"Checking the determinism of the input data")
```

Comment:
Create a separate method for this check?

Reply:
already created in `analyzeDataFrameDeterminism`